sdk/core/azure-core/src/main/java/com/azure/core/http/policy/BearerTokenAuthenticationPolicy.java

```diff
@@ -33,22 +36,78 @@
      */
     public BearerTokenAuthenticationPolicy(TokenCredential credential, String... scopes) {
         Objects.requireNonNull(credential);
-        Objects.requireNonNull(scopes);
-        assert scopes.length > 0;
         this.credential = credential;
         this.scopes = scopes;
-        this.cache = new SimpleTokenCache(() -> credential.getToken(new TokenRequestContext().addScopes(scopes)));
+        this.cache = new AccessTokenCache(credential);
+    }
+
+    /**
+     * Executed before sending the initial request and authenticates the request.
+     *
+     * @param context The request context.
+     * @return A {@link Mono} containing {@link Void}
+     */
+    public Mono<Void> authorizeRequest(HttpPipelineCallContext context) {
+        if (this.scopes == null) {
```

Comment:
Is it even valid to have `null` scopes? Should the constructor throw if `scopes` is null?

Reply:
It is possible depending on the SDK, e.g. the ARM challenge auth policy that will extend from this policy, it is possible for the scopes to be null, as it determines on the fly during request processing.